editor/src/components/canvas/controls/formula-bar.tsx

```diff
@@ -81,38 +88,60 @@ export const FormulaBar = betterReactMemo('FormulaBar', () => {
     [saveTimerRef, selectedElement, dispatchUpdate],
   )
 
+  const buttonsVisible = isFeatureEnabled('TopMenu ClassNames') && selectedElement != null
+  const classNameFieldVisible =
+    isFeatureEnabled('TopMenu ClassNames') && selectedElement != null && selectedMode === 'css'
+  const inputFieldVisible = !classNameFieldVisible
+
   return (
     <SimpleFlexRow
       style={{
         flexGrow: 1,
         height: UtopiaTheme.layout.inputHeight.default,
       }}
     >
-      <input
-        type='text'
-        css={{
-          paddingLeft: 4,
-          paddingRight: 4,
-          border: '0px',
-          width: '100%',
-          height: '100%',
-          backgroundColor: colorTheme.canvasBackground.value,
-          borderRadius: 5,
-          transition: 'background-color .1s ease-in-out',
-          '&:hover': {
-            '&:not(:disabled)': {
+      {buttonsVisible && <ModeToggleButton />}
+      {classNameFieldVisible && selectedMode === 'css' && (
```

Comment:
`classNameFieldVisible` already checks for `selectedMode === 'css'`

Reply:
oops, fixed